src/django_components/component.py

```diff
@@ -670,39 +669,37 @@ def _render_impl(
         cache_inlined_css(self.__class__, self.css or "")
 
         with _prepare_template(self, context, context_data) as template:
-            fills = self._escape_slot_fills(slots, escape_slots_content)
+            # For users, we expose boolean variables that they may check
+            # to see if given slot was filled, e.g.:
+            # `{% if variable > 8 and component_vars.is_filled.header %}`
+            is_filled = SlotIsFilled(slots_untyped)
```

Comment:
Since all slots and fills are now resolved at runtime (we no longer statically analyse the Django Template for Nodes), then it's no longer possible to define all available slots. Instead, we only know about the fills that the user provided.

So if we rendered a component like so:

```django
{% component "my_table" %}
  {% fill "header" %}
    Hello!
  {% endfill %}
{% endcomponent %}
```

Then, BEFORE,  `is_filled` dictionary would have been something like this:

```py
{
    "header": True,
    "main": False,
    "footer": False,
}
```

But NOW, the `is_filled` dictionary does NOT have the falsy values:

```py
{
    "header": True,
}
```

So, to keep the behavior consistent, I added a `SlotIsFilled` class, which behaves like a regular dictionary, but returns False on missing key.

So the difference is minor, but what was previously an invalid key (and coerced by Django into empty string) will now return False:

Before:
```django
{{ component_vars.is_filled.header }} -> True
{{ component_vars.is_filled.footer }} -> False
{{ component_vars.is_filled.nonexist }} -> "" (invalid key)
```

After:
```django
{{ component_vars.is_filled.header }} -> True
{{ component_vars.is_filled.footer }} -> False
{{ component_vars.is_filled.nonexist }} -> False
```